content/recipes/authoring/kata-snippets/submission-tests.md

```diff
@@ -0,0 +1,88 @@
+- **Test cases should be independent**, they should not rely on any state passed from one test case to another.
+
+
+## Reference Solution
+
+Some test suites require a reference solution to generate the expected value(s) and compare them to those returned by the user solution. While such practice is common and sometimes necessary, there are things which have to be handled carefully:
+
+- **Avoid using a reference solution at all if possible.** Sometimes there's no other way, but quite often a reference solution is simply not necessary: test cases can be generated with the answer known upfront, which eliminates an entire class of potential problems associated with the use of reference solutions:
+  - Extra time spent on running a reference solution
+  - Reference solution being accessible to users by mistake
+  - Input mutation by the user solution which can affect the input passed to the reference solution, or make assertion messages confusing
+  - Incorrect implementation of the reference solution leading to rejection of valid user solutions
+- **The reference solution should not be revealed to the user.** When an assertion fails or the test suite crashes, some testing frameworks print fragments of source code which caused the failure to the console. It may happen that such printed failure messages or stack traces expose information about the solution which should not be revealed.
+- **The reference solution shouldn't be accessible to the user solution.** It should not be possible to call the reference solution directly, or implement the user solution as an alias or wrapper around the reference solution. The reference solution should be completely inaccessible outside the submit tests. In particular, it should not be a global and/or public function. Check the [reference page for your language](/languages/) to see how to prevent this problem in your tests.
+
+
+## Fixed Tests
+
+Fixed tests are tests with predetermined inputs and outputs, and do not change between test runs.
+
+- **Each requirement should have a corresponding fixed test.** Every aspect of the specification in the kata description should be explicitly tested with at least one, possibly more, dedicated and properly labeled assertion(s). Fixed tests are usually easier to debug, so ideally a fixed test should fail before the corresponding scenario is tested with random tests.
+- **Tests should check the solution with edge cases and cases that require special handling.** For example, empty arrays and empty strings should be covered by fixed tests unless the kata description _explicitly_ states that such inputs need not be considered.
+
+
+## Random Tests
+
+Random tests are uncommon in "real life" coding and are somewhat specific to Codewars. They are required by many kata to reject solutions based on input probing, hard-coding, and other workarounds. The goal of random tests is to make the inputs unpredictable so that only proper solutions may pass.
```

Comment:
```suggestion
Random tests are uncommon in "real life" coding and are somewhat specific to Codewars. They are required to reject invalid approaches based on input probing, hard-coding, and other workarounds. The goal of random tests is to make the inputs unpredictable so that only solutions that are actually solving the task may pass.
```

Reply:
LGTM